Kernel/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-add_compile_options(-Os)
+add_compile_options(-Os -ggdb3)
```

Comment:
Just a reminder, still  probably need to make this conditional based on; 

```
commit a731ccd4a0bd55dd1a7a320c4c6e9bc383cb2618
Author: Andreas Kling <kling@serenityos.org>
Date:   Sat Feb 22 18:39:22 2020 +0100

    Kernel: Build without debugging symbols by default

    Compiling with -g adds roughly 30% to kernel build times. Anyone who
    wants this can turn it on locally instead.

commit 46c60fd451e478ee0bc834de7ff129d85d437bb3
Author: Brian Gianforcaro <b.gianfo@gmail.com>
Date:   Sun Jan 12 22:50:47 2020 -0800

    Debugging: Add kernel debugging support

    Introduce the 'debug-kernel' script to allow developers to
    quickly attach a debugger to the QEMU debug remote. The
    setting (-s) is already enabled by ./run today when using
    QEMU for virtualisation.

    If the system is running under QEMU, the debugger
    will break in when the script is run. If you add
    the -S option to QEMU it will wait for the debugger
    to connect before booting the kernel. This allows
    you to debug the init/boot process.

    Personally I use cgdb instead of gdb, so I opted
    to make the debugger used by the script customizable
    via an environment variable.

    This change also adds -g3 to the kernel build so that
    rich debug symbols are available in the kernel binary.
```

Reply:
Ah yes, I thought I had removed that for now. Turns out I re-added it to test my other changes. :+1:

Reply:
It would be interesting to know how much of a regression this really is these days, considering when @awesomekling  reverted the change we weren't even compiling with optimizations enabled and were probably using a 9 series gcc.

Reply:
Here's the current numbers from my machine.

Without;  
```
Without: 
$ time ninja Kernel
[0/2] Re-checking globbed directories...
[259/259] Linking CXX executable Kernel/Kernel

real	0m47.773s
user	5m57.104s
sys	0m11.670s
```
With `-gddb3`:
```
$ time ninja Kernel
[0/2] Re-checking globbed directories...
[259/259] Linking CXX executable Kernel/Kernel

real	1m2.625s
user	7m44.211s
sys	0m16.173s
```

Reply:
Sounds like we'd probably want a cmake option for it. :)